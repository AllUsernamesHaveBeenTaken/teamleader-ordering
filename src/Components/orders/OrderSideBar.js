import React from 'react';
import styled from 'styled-components';

import OrderSideBarItem from './OrderSideBarItem';
import { OrderContext } from '../../Contexts';

const OrderSideBar = () => (
  <OrderContext.Consumer>
    {context => (
      <SideBarWrapper>
        {context.orders.map(order => (
          <OrderSideBarItem key={`order-${order.id}`} id={order.id} selectOrder={context.selectOrder} />
        ))}
      </SideBarWrapper>
    )}
  </OrderContext.Consumer>
);

export default OrderSideBar;

const SideBarWrapper = styled.div`
  box-shadow: 10px 0 5px -2px #00000022;
  height: 100vh;
`;
