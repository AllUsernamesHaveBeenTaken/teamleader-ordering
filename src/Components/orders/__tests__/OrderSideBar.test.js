import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import OrderSideBar from '../OrderSideBar';

afterEach(() => {
  cleanup();
  console.error.mockClear();
});

console.error = jest.fn();

const orders = [
  {
    id: '1',
    'customer-id': '1',
    items: [
      {
        'product-id': 'B102',
        quantity: '10',
        'unit-price': '4.99',
        total: '49.90',
      },
    ],
    total: '49.90',
  },
  {
    id: '2',
    'customer-id': '2',
    items: [
      {
        'product-id': 'B102',
        quantity: '5',
        'unit-price': '4.99',
        total: '24.95',
      },
    ],
    total: '24.95',
  },
];

const selectOrder = jest.fn();

// OrderSideBar's props = orders, selectOrder
test('<OrderSideBar /> with orders', () => {
  const { getAllByTestId, queryByTestId } = render(
    <OrderSideBar orders={orders} selectOrder={selectOrder} />,
  );

  expect(getAllByTestId('menu-item')).length = orders.length;

  expect(queryByTestId('no-orders')).toBeFalsy();
});

test('<OrderSideBar /> with outorders', () => {
  const { getByTestId, queryAllByTestId } = render(
    <OrderSideBar selectOrder={selectOrder} />,
  );

  expect(queryAllByTestId('menu-item')).length = 0;

  expect(getByTestId('no-orders')).toBeTruthy();
});
