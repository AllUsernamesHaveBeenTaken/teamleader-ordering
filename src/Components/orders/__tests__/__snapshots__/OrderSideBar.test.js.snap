// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderSideBar /> with orders 1`] = `
<div
  class="sc-htpNat bFasjL"
>
  <div
    class="sc-bdVaJa hLZleQ"
    data-testid="menu-item"
  >
    <h1
      class="sc-bwzfXH hDweOk"
      data-testid="menu-item-title"
    >
      Order No. 1
    </h1>
  </div>
  <div
    class="sc-bdVaJa hLZleQ"
    data-testid="menu-item"
  >
    <h1
      class="sc-bwzfXH hDweOk"
      data-testid="menu-item-title"
    >
      Order No. 2
    </h1>
  </div>
</div>
`;
